src/user_activate_card_workflow.js

```diff
@@ -0,0 +1,123 @@
+import http from 'k6/http';
+import { sleep } from 'k6';
+import { check } from 'k6';
+import { generateFakeFiscalCode } from './modules/helpers.js';
+
+export let options = {
+    vus: 1,
+    stages: [
+        { duration: '30s', target: 1 }, // simulate ramp-up of traffic from 1 to 100 users over 5 minutes.
+    ],
+    thresholds: {
+        http_req_duration: ['p(99)<1500'], // 99% of requests must complete below 1.5s
+        'http_req_duration{pagoPaMethod:ActivateNewCard}': ['p(95)<1000'], // threshold on API requests only
+        'http_req_duration{pagoPaMethod:GetCGNActivation}': ['p(95)<1000'], // threshold on API requests only
+        'http_req_duration{pagoPaMethod:GetCGNStatus}': ['p(95)<1000'], // threshold on API requests only
+        'http_req_duration{pagoPaMethod:GetEycaStatus}': ['p(95)<1000'], // threshold on API requests only
+    },
+};
+
+export default function () {
+    // Values from env var.
+    var urlBasePath = `${__ENV.BASE_URL}`
+    var funcKey = `${__ENV.FUNC_KEY}`
+
+    // Born in the ninenties.
+    var fiscalCode = generateFakeFiscalCode("9");
+    console.log('Fiscal code: ' + fiscalCode)
+    var headersParams = {
+        headers: {
+            'Content-Type': 'application/json',
+            'x-functions-key': funcKey,
+        },
+    };
+
+    // Activate new card.
+    var tag = {
+        pagoPaMethod: "ActivateNewCard",
+    };
+    var url = `${urlBasePath}/api/v1/cgn/${fiscalCode}/activation`;
+    var payload = JSON.stringify({});
+    var r = http.post(url, payload, headersParams, {
+        tags: tag,
+    });
+    console.log("Activate new card Fiscal code " + fiscalCode + " Status " + r.status);
+    check(r, { 'status is 201': (r) => r.status === 201 }, tag);
+
+    //Simulating mobile app polling to check activation status.
+    var maxRetries = 20;
+    var retryCount = 0;
+    var cardActivated = false;
+    do {
+        sleep(1)
+        console.log('Polling Get CGN Activation: ' + r.status + ' retry: ' + (retryCount + 1));
+
+        // Get CGN Activation (200 with one of the following statuses: completed, pending, running)
+        tag = {
+            pagoPaMethod: "GetCGNActivation",
+        };
+        url = `${urlBasePath}/api/v1/cgn/${fiscalCode}/activation`;
+        r = http.get(url, headersParams, {
+            tags: tag,
+        });
+        check(r, { 'Get CGN Activation is 200': (r) => (r.status === 200) },
+            tag
+        );
+
+        // If status is "completed" then the card is activated.
+        var jsonBody = JSON.parse(r.body);
```

Comment:
please check if _body_ is undefined first. In case the response is not as you might expect, _body_ could be undefined.

Refer to https://k6.io/docs/testing-guides/running-large-tests/#make-your-test-code-resilient

Reply:
Done!